File: 백준/Silver/1463. 1로 만들기/1로 만들기.swift
let n = Int(readLine()!)!
var d = Array(repeating: 0, count: 1000001)
if (n > 1) {
    for i in 2...n {
        d[i] = d[i-1] + 1
        if (i%2 == 0) { d[i] = [d[i], d[i/2]+1].min()! }
        if (i%3 == 0) { d[i] = [d[i], d[i/3]+1].min()! }
    }
}
print(d[n])

File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        int[] d = new int[N+1];
        d[1] = 0;
        for(int i=2; i<=N; i++) {
            d[i] = d[i-1] + 1;
            if(i%2 == 0) d[i] = Math.min(d[i], d[i/2]+1);
            if(i%3 == 0) d[i] = Math.min(d[i], d[i/3]+1);
        }
        System.out.println(d[N]);
    }
}